### PR TITLE
Refactor: Idiomatic React Query

### PR DESCRIPTION
**Pull request by Bushuo**
## Basics

- [x] I added a line to [changelog.md](/doc/changelog.md)
- [X] The PR is rebased with current master.
- [X] Details of what you changed are in commit messages.
- [x] References to issues, e.g. `close #X`, are in the commit messages and changelog.
- [ ] The buildserver is happy.

## Checklist

- [x] I have installed and I am using [pre-commit hooks](../doc/contrib/README.md#Hooks)
- [x] I fully described what my PR does in the documentation
- [x] I fixed all affected documentation
- [ ] I fixed the introduction tour
- [ ] I wrote migrations in a way that they are compatible with already present data
- [ ] I fixed all affected decisions
- [ ] I added automated tests or a [manual test protocol](../doc/tests/manual/protocol.md)
- [x] I added code comments, logging, and assertions as appropriate
- [ ] I translated all strings visible to the user
- [ ] I mentioned [every code or binary](https://github.com/ElektraInitiative/PermaplanT/blob/master/.reuse/dep5) not directly written or done by me in [reuse syntax](https://reuse.software/)
- [ ] I created left-over issues for things that are still to be done
- [x] Code is conforming to [our Architecture](/doc/architecture)
- [x] Code is conforming to [our Guidelines](/doc/guidelines)
- [x] Code is consistent to [our Design Decisions](/doc/decisions)
- [ ] Exceptions to any guidelines are documented

## Review

- [ ] I've tested the code
- [ ] I've read through the whole code
- [ ] I've read through the whole documentation
- [ ] I've checked conformity to guidelines

### CODE REVIEW

**Comment by Bushuo** [comment]
jenkins build please

**Comment by Bushuo** [comment]
jenkins build please

**Review by markus2330 — APPROVED** [review]
LGTM. @danielsteinkogler pls review

**Comment by markus2330** [comment]
jenkins build please

**Comment by markus2330** [comment]
jenkins build please

**Comment by markus2330** [comment]
jenkins build please

**Comment by Bushuo** [comment]
@markus2330 @4ydan
Somehow the e2e tests fails for the inventory.
I am pretty sure that the e2e test is wrong in this case.
locally creating, editing and archiving works fine.

**Comment by 4ydan** [comment]
It seems the tests fails because the database is not properly reset. (Failing because map/seed already exists).

File: doc/guidelines/frontend-api-calls.md [hunk 1]
```diff
@@ -86,3 +89,76 @@ const addFile = useMutation({
     },
   });
 ```
+
+### Scalable Approach
```
Comment (danielsteinkogler) [inline doc/guidelines/frontend-api-calls.md:93]: I really like this approach. It also gives me a better understanding of the advantages of using useQuery.

File: frontend/src/features/map_planning/hooks/mapEditorHookApi.ts [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+
+/**
+ * Get map data for the given map id.
+ */
+export function useMap(mapId: number) {
+  return useQuery({
+    queryKey: MAP_EDITOR_KEYS.map(mapId),
+    queryFn: getMapQueryFn,
+    refetchOnWindowFocus: false,
+    // TODO: add error message
```
Comment (danielsteinkogler) [inline frontend/src/features/map_planning/hooks/mapEditorHookApi.ts:64]: is there already a left-over-issue that could be linked?

File: frontend/src/features/map_planning/hooks/tourHookApi.ts [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+/**
+ * Re-enable the guided tour.
+ */
+export function useReenableTour() {
+  const queryClient = useQueryClient();
+
+  return useMutation({
+    mutationFn: () => updateTourStatus({ editor_tour_completed: false }),
+    onError: () => {
+      // TODO toast
```
Comment (danielsteinkogler) [inline frontend/src/features/map_planning/hooks/tourHookApi.ts:40]: i think we should also create a left-over-issue for missing toast messages

**Review by danielsteinkogler — COMMENTED** [review]
Great job! The architecture is now much cleaner with query hooks centralized for each feature. I'm just not sure how we typically address TODOs in the code. I believe it would make sense to create a list of leftover issues to ensure we don't forget to resolve them.

**Comment by Bushuo** [comment]
@danielsteinkogler 
Thanks for the review, glad that you like the approach.
I created #1136 for the leftover error messages.

**Comment by Bushuo** [comment]
- 1: considering it as passed. The video shows there was a problem with the backend...

**Comment by Bushuo** [comment]
- 2: passed

**Comment by Bushuo** [comment]
jenkins build please

**Comment by Bushuo** [comment]
jenkins build please

**Pull request merged**